lib/parser/openmp-grammar.h

```diff
@@ -439,18 +439,18 @@ TYPE_PARSER(construct<OpenMPAtomicConstruct>(Parser<OmpAtomicRead>{}) ||
     construct<OpenMPAtomicConstruct>(Parser<OmpAtomic>{}))
 
 // OMP CRITICAL
-TYPE_PARSER(startOmpLine >> "END CRITICAL"_tok >>
-    construct<OmpEndCritical>(maybe(parenthesized(name))) / endOmpLine)
-
-TYPE_PARSER(
-    sourced(construct<OpenMPCriticalConstructDirective>(
-        "CRITICAL" >> maybe(parenthesized(name)),
-        maybe("HINT" >> construct<OpenMPCriticalConstructDirective::Hint>(
-                            parenthesized(constantExpr))))) /
+TYPE_PARSER(startOmpLine >>
+    sourced(construct<OmpEndCriticalDirective>(
+        verbatim("END CRITICAL"_tok), maybe(parenthesized(name)))) /
```

Comment:
If the space is required in `END CRITICAL`, use `_sptok`, not `_tok`.

Reply:
Thanks! Based on the OpenMP spec, here the space is optional.